csrc/executor_kernel_arg.cpp

```diff
@@ -298,6 +282,64 @@ std::vector<std::byte> polymorphicValueToBytes(
           dtype,
           " type: only complex float and complex double are supported.");
     }
+  } else if (argument.is<StructHandle>()) {
+    // FUSER_PERF_SCOPE("polymorphicValueToBytes(StructHandle)");
+    std::vector<std::byte> buffer;
+    const auto& dtype_ = std::get<StructType>(dtype.type);
+    auto& data = argument->*&TensorMetaData::data;
+    auto& logical_size = argument->*&TensorMetaData::logical_size;
+    auto& alloc_stride = argument->*&TensorMetaData::alloc_stride;
+    if (argument.as<StructHandle>().is<TensorMetaData>()) {
+      // special handle for TensorMetaData so that CPU overhead is minimal.
+      if (index_type == PrimDataType::Int) {
```

Comment:
Why can't this first path be used for the latter path with `sizeof(int64_t)` just replaced with `sizeof(int)`?

Reply:
For the `index_type == PrimDataType::Int` path, we are just copying bytes. However, for the `index_type == PrimDataType::Int32` path, we need an additional
```C++
        std::vector<int32_t> logical_size32(
            logical_size.begin(), logical_size.end());
```
which does `int64_t -> int32_t` casting.